packages/core/src/lib/providers.ts

```diff
@@ -96,9 +96,14 @@ function normalizeEndpoint(
   // NOTE: This need to be checked when constructing the URL
   // for the authorization, token and userinfo endpoints.
   const url = new URL(e?.url ?? "https://authjs.dev")
-  for (const k in e?.params) {
-    if (e?.params && k === "claims") e.params[k] = JSON.stringify(e.params[k])
-    url.searchParams.set(k, e?.params[k])
+  if (e?.params) {
+    for (const k in e.params) {
```

Comment:
I don't think this change is necessary, `for..of` will only iterate if `e?.params` is an iterable object, which `undefined` isn't.

Reply:
This check tells typescript that `e` is defined on lines 102 and 104.

Reply:
It's unnecessary as per my comment. You cannot enter the for loop where `e.params` is not defined.

Reply:
hmm. current source code might suggest otherwise https://github.com/nextauthjs/next-auth/blob/326eadf0edad2a49a92545df44623304cc521958/packages/core/src/lib/providers.ts#L100

Will double check when on a computer (I believe TS should be smart enough here), but still, I would like to take a closer look at where this is causing an OOM issue. 🤔